doc/ERROR_CONCEPT.md

```diff
@@ -0,0 +1,250 @@
+
+With this line 5/6 can still be present but in a much more efficient and succinct way.
+
+### Error/Warning/Fatal etc. splitting in the specification file
+
+I would suggest to remove this concept from the specification and let the developer use a respective method to indicate if it is a warning/error/etc.
+The macros ELEKTRA_SET_ERROR, ELEKTRA_ADD_WARNING ... should be enough to make developers cautious of wrong method usage.
+What I have seen is that the return code is the only matter if it is an error or not.
+
+### Additional information which is not present
+
+Some errors can also guide users to fix the error by suggesting a possible solution:
+* Retry as sudo
+* Fix file permissions
+* Check if the file exists
+* etc.
+Developers though have to be very careful when suggesting solutions because it can easily mislead users. For many cases though, 
+this can help users a lot.
+I would suggest an optional extra line (which is only printed if the developer adds a metafield `solution` to the parentkey):
+`Possible Solution: ...`
+
+# Conclusion
+
+With the new concept we would not require a `specification` file at all which helps developers immensely but also keeps the
+most relevant information for the users. An error like this:
+
+```
+1. The command kdb set failed while accessing the key database with the info:
+2. Sorry, the error (#121) occurred ;(
+3. Description: validation failed
+4. Reason: Validation of key "<key>" with string "<value>" failed.
+5. Ingroup: plugin
+6. Module: enum
+7. At: ....../src/plugins/enum/enum.c:218
+8. Mountpoint: <parentKey>
+9. Configfile: ...../<file>.25676:1549919217.284067.tmp
+```
+
+would be changed to this:
+```
+1. Plugin enum issued a error while accessing the key database with the info:
+2. Validation of key "<key>" with string "<value>" failed.
+```
+
+with optionally a third line indicating a solution. Eg. for a permission related error there would be a third line:
+```
+3. Possible Solution: Retry the command as sudo (sudo !!)
+```
+
+Benefits:
+* Much more relevant information in the error message
+* Ease of development:
+no merge conflicts, no extra line in the `specification` file leading to chaos (such as it is now)
+New plugins will not have to choose an existent error category or implement a new one which could potentially be wrong
+* Less maintenance effort:
+No consolidation, splitting, etc. of categories needed in future versions
+No need for splitting the `specification` file because it is too large
+
+Drawbacks:
+* Possible small information loss by removing the description, `At` and `mountpoint`:
+`At` could be kept for devs by extending the `kdb set` command with an option `-v`, `--debug`, etc.
+The same applies for the `mountpoint`.
+Description comes with too many drawbacks to be kept
+* Possible misleading the user with the `Solution` line
+
+The benefits do outweigh the drawbacks. I would suggest an overloaded method like the following:
```

Comment:
overloading in C is a bit of an hack (needs macros), so better to give different names or simply return null pointers if there is no possible solution.